src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -618,17 +619,46 @@ await foreach (ConfigurationSetting setting in client.GetConfigurationSettingsAs
                 }).ConfigureAwait(false);
             }
 
-                foreach (var loadOption in _options.KeyValueSelectors)
+            foreach (KeyValueSelector loadOption in _options.KeyValueSelectors)
+            {
+                IAsyncEnumerable<ConfigurationSetting> settingsEnumerable;
+
+                if (string.IsNullOrEmpty(loadOption.SnapshotName))
                 {
-                    var selector = new SettingSelector
+                    settingsEnumerable = client.GetConfigurationSettingsAsync(
+                        new SettingSelector
+                        {
+                            KeyFilter = loadOption.KeyFilter,
+                            LabelFilter = loadOption.LabelFilter
+                        },
+                        cancellationToken);
+                }
+                else
+                {
+                    ConfigurationSettingsSnapshot snapshot;
+
+                    try
                     {
-                        KeyFilter = loadOption.KeyFilter,
-                        LabelFilter = loadOption.LabelFilter
-                    };
+                        snapshot = await client.GetSnapshotAsync(loadOption.SnapshotName).ConfigureAwait(false);
+                    }
+                    catch (RequestFailedException)
+                    {
+                        throw new InvalidOperationException($"Could not load snapshot with name '{loadOption.SnapshotName}'");
+                    }
```

Comment:
That was the point, should we check for that specifically instead? We could see if it's a 404 and otherwise just rethrow the original exception?

Reply:
> We could see if it's a 404 and otherwise just rethrow the original exception?

Indeed. We should only catch an exception representing the error we want to handle. RequestFailedException is the propagation mechanism for other types of errors (e.g. authentication or authorization failure) that we don't want to catch here.